### PR TITLE
Replace select with poll to avoid the upper limit of 1024 file descri…

### LINKED ISSUE #5: segfault on TCP mode after set clients number on some value
I am trying to use dnsperf (build on Fedora 24) against [CoreDNS](https://github.com/coredns/coredns) to figure out how many connection it can open.
File dnsperf.c has line
`#define MAX_SOCKETS 256`
so I changed it with
`#define MAX_SOCKETS 2048`
Observed that command
`dnsperf -s 127.0.0.1 -p 53 -d dns.txt -Q 10000 -l 10 -z -c 1077`
works fine but command
`dnsperf -s 127.0.0.1 -p 53 -d dns.txt -Q 10000 -l 10 -z -c 1078`
failed with message "Segmentation fault".

### PR DESCRIPTION
**Pull request by saradickinson**
…ptors imposed by select

Increase default limit on the number of clients from 256 to 2048
Should fix https://github.com/Sinodun/dnsperf-tcp/issues/5

### CODE REVIEW

File: os.c [hunk 1]
```diff
@@ -138,40 +134,30 @@ isc_result_t
 perf_os_waituntilanyreadable(int *fds, unsigned int nfds, int pipe_fd,
 			     isc_int64_t timeout)
 {
-	fd_set read_fds;
+	struct pollfd read_fds[nfds];
```
Comment (UladzimirTrehubenka) [inline os.c:137]: How do you build this code (nfds here is not constant)?
Would be better to use
`perf_os_waituntilanyreadable(struct pollfd *read_fds, unsigned int nfds, isc_int64_t timeout)`
instead. In the code we have chain `tinfo->socks -> fds -> read_fds` and we can avoid using intermediate fds array here.
Also in dnsperf.c there are no check `perf_os_waituntilanyreadable` ret value - so this check also can be removed from the function.

File: os.c [hunk 2]
```diff
@@ -76,25 +76,23 @@ perf_os_handlesignal(int sig, void (*handler)(int))
...
-		tvp = NULL;
+		timeout_msec = 0;
 	} else {
-		tv.tv_sec = timeout / MILLION;
-		tv.tv_usec = timeout % MILLION;
-		tvp = &tv;
+		timeout_msec = timeout / THOUSAND;
 	}
-	n = select(fd + 1, NULL, &write_fds, NULL, tvp);
+	n = poll((struct pollfd*)&write_fds, 1, timeout_msec);
```
Comment (banburybill) [inline os.c]: I'm curious about why the cast is necessary. Surely

poll(write_fds, 1, timeout_msec);

should be OK?

File: os.c (hunk 1 above)
Comment (banburybill) [reply os.c:137]: struct pollfd read_fds[nfds]; requires C99. Perhaps configure.ac should check for this.

The implementation of perf_os_waituntilanyreadable() uses select internally, and so has the same limitation on number of sockets. Possibly reimplementing that to use poll() and then using that makes sense.

Comment (banburybill) [reply os.c:137]: In configure.ac, s/AC_PROG_CC/AC_PROG_CC_C99/ for first.

**Pull request merged**